src/applications/model/gaming-streaming-server.cc

```diff
@@ -95,29 +97,37 @@ GamingStreamingServer::GetTotSentPackets (void)
   return m_totSentPackets;
 }
 
+uint32_t
+GamingStreamingServer::GetTotSentBytes (void)
+{
+  NS_LOG_FUNCTION (this);
+  return m_totSentBytes;
+}
+
 void
 GamingStreamingServer::EraseStatistics (void)
 {
   NS_LOG_FUNCTION (this);
   m_totSentPackets = 0;
+  m_totSentBytes = 0;
 }
 
 void
 GamingStreamingServer::AddNewTraffic (Ptr<RandomVariableStream> packetSize,
                                       Ptr<RandomVariableStream> interArrivalTime)
 {
-  NS_LOG_FUNCTION (this);
-  m_pksList.push_back (packetSize);
-  m_iatList.push_back (interArrivalTime);
+  NS_LOG_FUNCTION (this << packetSize << interArrivalTime);
+  m_packetSizeVariables.push_back (packetSize);
+  m_interArrivalTimesVariables.push_back (interArrivalTime);
+  m_sendEvent.push_back (EventId ());
 }
 
 void
-GamingStreamingServer::Send (Ptr<RandomVariableStream> packetSize,
-                             Ptr<RandomVariableStream> interArrivalTime)
+GamingStreamingServer::Send (int idx)
 {
   NS_LOG_FUNCTION (this);
 
-  uint32_t pktSize = packetSize->GetInteger ()+12; //TODO: think about 12
+  uint32_t pktSize = std::max ((int)m_packetSizeVariables[idx]->GetInteger (), 12); //TODO: think about 12
```

Comment:
separate `GetInteger` for easier debugging. not sure about that reference to 12, could you explain?